src/main/java/com/minecolonies/entity/ai/citizen/guard/EntityAIRangeGuard.java

```diff
@@ -0,0 +1,300 @@
+            worker.setCanPickUpLoot(true);
+        }
+    }
+
+    @Override
+    protected AIState searchTarget()
+    {
+        if (checkOrRequestItems(new ItemStack(Items.BOW)))
+        {
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+        worker.setHeldItem(worker.findFirstSlotInInventoryWith(Items.BOW));
+        return super.searchTarget();
+    }
+
+    /**
+     * Follow the target and kill it.
+     *
+     * @return the next AIState.
+     */
+    protected AIState huntDown()
+    {
+        if (!targetEntity.isEntityAlive() || checkOrRequestItems(new ItemStack(Items.BOW)))
+        {
+            targetEntity = null;
+        }
+
+        if (targetEntity != null)
+        {
+            if (worker.getEntitySenses().canSee(targetEntity) && worker.getDistanceToEntity(targetEntity) <= MAX_ATTACK_DISTANCE)
+            {
+                worker.resetActiveHand();
+                attackEntityWithRangedAttack(targetEntity, DAMAGE_PER_ATTACK);
+                setDelay(getReloadTime());
+                attacksExecuted += 1;
+
+                if (attacksExecuted >= getMaxAttacksUntilRestock())
+                {
+                    return AIState.GUARD_RESTOCK;
+                }
+
+                return AIState.GUARD_HUNT_DOWN_TARGET;
+            }
+            worker.setAIMoveSpeed((float) (BASE_FOLLOW_SPEED + BASE_FOLLOW_SPEED_MULTIPLIER * worker.getExperienceLevel()));
+            worker.isWorkerAtSiteWithMove(targetEntity.getPosition(), MOVE_CLOSE);
+
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+
+        worker.setAIMoveSpeed(1);
+        return AIState.GUARD_SEARCH_TARGET;
+    }
+
+    @Override
+    public void attackEntityWithRangedAttack(@NotNull EntityLivingBase entityToAttack, float baseDamage)
+    {
+        EntityTippedArrow arrowEntity = new EntityTippedArrow(this.worker.worldObj, worker);
+        double xVector = entityToAttack.posX - worker.posX;
+        double yVector = entityToAttack.getEntityBoundingBox().minY + entityToAttack.height / AIM_HEIGHT - arrowEntity.posY;
+        double zVector = entityToAttack.posZ - worker.posZ;
+        double distance = (double) MathHelper.sqrt_double(xVector * xVector + zVector * zVector);
+
+        //Lower the variable higher the chance that the arrows hits the target.
+        double chance = HIT_CHANCE_DIVIDER / (worker.getExperienceLevel() + 1);
+
+        arrowEntity.setThrowableHeading(xVector, yVector + distance * AIM_SLIGHTLY_HIGHER_MULTIPLIER, zVector, (float) ARROW_SPEED, (float) chance);
+
+        addEffectsToArrow(arrowEntity, baseDamage);
+
+        worker.addExperience(XP_EACH_ARROW);
+        worker.faceEntity(entityToAttack, (float) TURN_AROUND, (float) TURN_AROUND);
+        worker.getLookHelper().setLookPositionWithEntity(entityToAttack, (float) TURN_AROUND, (float) TURN_AROUND);
+
+        double xDiff = targetEntity.posX - worker.posX;
+        double zDiff = targetEntity.posZ - worker.posZ;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'zDiff' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)